tensorflow/compiler/xla/service/gpu/tree_reduction_rewriter.cc

```diff
@@ -256,16 +261,23 @@ class ReductionRewriterVisitor : public DfsHloRewriteVisitor {
     VLOG(1) << "Generated: " << out->ToString();
     return ReplaceWithNewInstruction(hlo, std::move(out));
   }
-
-  se::CudaComputeCapability cuda_compute_capability_;
+#if GOOGLE_CUDA
+    se::CudaComputeCapability cuda_compute_capability_;
+#elif TENSORFLOW_USE_ROCM
+    se::RocmComputeCapability rocm_compute_capability_;
+#endif
 };
 
 StatusOr<bool> GpuTreeReductionRewriter::Run(
     HloModule *module,
     const absl::flat_hash_set<absl::string_view> &execution_threads) {
   VLOG(5) << "Rewriter input: " << module->ToString();
   TF_ASSIGN_OR_RETURN(bool changed,
+#if GOOGLE_CUDA
                       ReductionRewriterVisitor(cuda_compute_capability_)
+#elif TENSORFLOW_USE_ROCM
```

Comment:
This is causing the issue with the CPU test. When CUDA or ROCM are not defined we end up with a dangling RunOnModule()

Reply:
Fixed it. Please check.